### ISSUE TITLE
clippy::op_ref false positive with deref coercions

### ISSUE THREAD

**Issue by Manishearth** [rust-lang/rust-clippy#4556]
```rust
    let a = "a".to_string();
    let b = "a";

    if b < &a {
        println!("OK");
    }
```

This warns, but it should not, since the `&a` is necessary to coerce the type. We even incorrectly test for this case.

**Comment by mikerite** [comment]
Is this the same as  #2597?

**Comment by Manishearth** [comment]
Yep
